app/src/test/java/com/codesoom/assignment/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,244 @@
+
+        @Nested
+        @DisplayName("등록될 user의 name이 빈값이라면")
+        class Context_with_user_with_empty_name {
+
+            UserData givenInvalidUserData;
+
+            @BeforeEach
+            void prepare() {
+                givenInvalidUserData = UserData.builder()
+                        .password("!234")
+                        .email("pjh0819@naver.com")
+                        .build();
+            }
+
+            @Test
+            @DisplayName("400(Bad Request)를 응답합니다.")
+            void it_return_bad_request() throws Exception {
+                mockMvc.perform(post("/user")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(userDataToContent(givenInvalidUserData)))
+                        .andExpect(status().isBadRequest())
+                        .andDo(print());
+            }
+        }
+
+        @Nested
+        @DisplayName("등록될 user의 password가 빈값이라면")
+        class Context_with_user_with_empty_password {
+
+            UserData givenInvalidUserData;
+
+            @BeforeEach
+            void prepare() {
+                givenInvalidUserData = UserData.builder()
+                        .name("Hyuk")
+                        .email("pjh0819@naver.com")
+                        .build();
+            }
+
+            @Test
+            @DisplayName("400(Bad Request)를 응답합니다.")
+            void it_return_bad_request() throws Exception {
+                mockMvc.perform(post("/user")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(userDataToContent(givenInvalidUserData)))
+                        .andExpect(status().isBadRequest())
+                        .andDo(print());
+            }
+        }
+
+        @Nested
+        @DisplayName("등록될 user의 email의 형식이 잘못되었다면")
+        class Context_with_user_with_invalid_email {
+
+            UserData givenInvalidUserData;
+
+            @BeforeEach
+            void prepare() {
+                givenInvalidUserData = UserData.builder()
+                        .name("Hyuk")
+                        .password("!234")
+                        .email("pjh0819")
+                        .build();
+            }
+
+            @Test
+            @DisplayName("400(Bad Request)를 응답합니다.")
+            void it_return_bad_request() throws Exception {
+                mockMvc.perform(post("/user")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(userDataToContent(givenInvalidUserData)))
+                        .andExpect(status().isBadRequest())
+                        .andDo(print());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("POST /user/{id} 요청은")
+    class Describe_post_id{
+
+        @Nested
+        @DisplayName("수정할 id와 user가 주어진다면")
+        class Context_with_id_and_user {
+
+            UserData givenUserData;
+            Long givenId= 1L;
```

Comment:
이 값은 수정되면 안되니까 이렇게 해줍시다.
```suggestion
            final Long givenId= 1L;
```

Reply:
네 !! final 추가하겠습니다